security/security-api/built-in-db-identity-store/README.md

```diff
@@ -0,0 +1,251 @@
+                // this method.
+                return httpMessageContext.notifyContainerAboutLogin(
+                    result.getCallerPrincipal(), result.getCallerGroups());
+            } else {
+                return httpMessageContext.responseUnauthorized();
+            }
+        }
+
+        return httpMessageContext.doNothing();
+    }
+
+}
+```
+Following snippet from `TestAuthenticationMechanism` delegates the validation task to underlying `DefaultIdentityStoreHandler`, which in turn validates the credentials against `DatabaseIdentityStore` as defined in this application.
+
+```java
+CredentialValidationResult result = identityStoreHandler.validate(
+    new UsernamePasswordCredential(name, password));
+```
+
+On successful authorization, container gets notified with caller details.
+
+```java
+return httpMessageContext.notifyContainerAboutLogin(
+    result.getCallerPrincipal(), result.getCallerGroups());
+```
+If credentials are not valid, an unauthorized response is sent back to the caller with status `401` indicating that authentication has failed with provided credentails.
+```java
+return httpMessageContext.responseUnauthorized();
+```
+
+When a request is made to the application, the roles the user is in, gets returned as part of the repsonse.
+
+```java
+@DeclareRoles({ "foo", "bar", "kaz" })
+@WebServlet("/servlet")
+public class Servlet extends HttpServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String webName = null;
+        if (request.getUserPrincipal() != null) {
+            webName = request.getUserPrincipal().getName();
+        }
+
+        response.getWriter().write("web username: " + webName + "\n");
+
+        response.getWriter().write("web user has role \"foo\": " + request.isUserInRole("foo") + "\n");
+        response.getWriter().write("web user has role \"bar\": " + request.isUserInRole("bar") + "\n");
+        response.getWriter().write("web user has role \"kaz\": " + request.isUserInRole("kaz") + "\n");
+    }
+
+}
+```
+Notice that, the container itself needs to be made aware of the supported roles and it is done with the help of `@DeclareRoles` annotation as shown above.
+
+```java
+@DeclareRoles({ "foo", "bar", "kaz" })
+```
+In Glassfish 5.0, role to group mapping is enabled by default. Hence there is no need to bundle `web.xml` with the application to provide mapping between roles and the groups in Glassfish.
+
+In this example, we would be using credentials of user `reza` to make a request and see if response is according to credentails/roles defined in `DatabaseSetup.java` above.
+
+**Steps:**
+
+* Since we are using default datasource bundled with Glassfish for `DatabaseIdentityStore`, we need to start the default database by running following command:
```

Comment:
Since we are using the default datasource bundled with GlassFish for `DatabaseIdentityStore`, start the default database by running the following command: